bitwarden_access_manager.py

```diff
@@ -0,0 +1,672 @@
+#!/usr/bin/env python3
+
+"""
+Bitwarden Access Manager
+
+CONFIGURATION
+
+[[member]]
+member_id = "2564c11f-fc1b-4ec7-aa0b-afaf00a9e4a4"
+member_name = "yan"
+email = "yan.68@hotmail.fr"
+type = 2 # member
+access_all = false
+groups = ["group1", "group2"]
+
+[[member]]
+member_id = "856cba2d-cae1-40e7-96cc-afaf00a8a4cb"
+member_name = "yunkel"
+email = "yunkel68@hotmail.fr"
+type = 0 # owner
+access_all = true
+groups = ["group1"]
+
+[[group]]
+group_id = "c6a13b93-edc1-4c3b-9fc5-afaf00a8d33f"
+group_name = "group1"
+access_all = false
+
+[[group]]
+group_id = "39b48ab2-81fd-40eb-87e9-afb0000110f3"
+group_name = "group2"
+access_all = false
+
+[[collection]]
+collection_id = "50351c20-55b4-4ee8-bbe0-afaf00a8f25d"
+external_id = "collection1"
+member_access = [
+  { member_id = "2564c11f-fc1b-4ec7-aa0b-afaf00a9e4a4", member_name = "yan", group = "c6a13b93-edc1-4c3b-9fc5-afaf00a8d33f" },
+  ]
+group_access = [
+  { group_id = "c6a13b93-edc1-4c3b-9fc5-afaf00a8d33f", group_name = "group1", read_only = true },
+  { group_id = "39b48ab2-81fd-40eb-87e9-afb0000110f3", group_name = "group2", read_only = true },
+]
+
+[[collection]]
+collection_id = "8e69ce49-85ae-4e09-a52c-afaf00a90a3f"
+external_id = ""
+member_access = [
+  { member_id = "2564c11f-fc1b-4ec7-aa0b-afaf00a9e4a4", member_name = "yan", group = "c6a13b93-edc1-4c3b-9fc5-afaf00a8d33f" },
+]
+group_access = [
+  { group_id = "c6a13b93-edc1-4c3b-9fc5-afaf00a8d33f", group_name = "group1", read_only = false },
+]
+"""
+
+from __future__ import annotations
+from dataclasses import dataclass
+from difflib import SequenceMatcher
+
+import json
+import os
+import requests
+import sys
+import tomllib
+
+from http.client import HTTPSConnection, HTTPResponse
+from typing import (
+    Any,
+    Dict,
+    List,
+    Generic,
+    Optional,
+    Tuple,
+    NamedTuple,
+    Set,
+    Iterable,
+    Protocol,
+    TypeVar,
+)
+class Member(NamedTuple):
+    id: str
+    name: str
+    email: str
+    type: str
```

Comment:
We could make it a bit more strongly typed and define an enum for this. For an example of this: https://github.com/ChorusOne/github-access-manager/blob/6fc4782bd5bbccdc52193f25f454321dd1ea6251/main.py#L148-L150

This has two advantages:

* It makes it easier to see from the source code what values are allowed.
* We can make the program complain about bad inputs.

We can convert a string to an enum with just the constructor, e.g. `OrganizationRole("admin")` for the example above, and we can get the underlying string back with `.value`, i.e. `OrganizationRole.ADMIN.value == "admin"`.

Reply:
I guess this is overlapping your comment below, please check my answer under this comment.

> First we can define an enum like so:
```
class MemberType(Enum):
    OWNER = 0
    ADMIN = 1
    USER = 2
    MANAGER = 3
    CUSTOM = 4
```